src/lib/fsm/index.ts

```diff
@@ -0,0 +1,54 @@
+export default class fsm {
```

Comment:
naming convention: Class names with PascalCase 
naming conventions: Naming without acronyms and specifying functionality "DeterministicFiniteStateMachine" rather than "fsm"

export conventions: (We havent defined it yet, but I would vote for no default exports)